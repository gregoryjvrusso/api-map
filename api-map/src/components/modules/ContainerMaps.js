import React, { useState, useEffect } from 'react'
import GoogleMaps from '../elements/GoogleMaps'
import { geoCodeGenerator } from '../helpers/geocode'
import styled from 'styled-components'

const Title = styled.p`
  font-size: 2em;
  font-weight: bold;
  margin: 0;
`
const Text = styled.p`
  font-size: 1em;
  margin: 0em 0.5em;
`
const ContainerMaps = ({ data }) => {
  const [latMap, setLat] = useState('')
  const [lngMap, setLng] = useState('')
  useEffect(() => {
    geoCodeGenerator(data.cep, {setLat}, {setLng})
  }, [data])


  return (
    <div>
      <Title>{data.logradouro}</Title>
      <Text>{data.bairro}</Text>
      <Text>{`${data.localidade}-${data.uf}`}</Text>
      <Text>{data.cep}</Text>
      {lngMap !== '' && latMap !== '' && 
        <GoogleMaps 
          latMap={latMap} 
          lngMap={lngMap} 
      />}
    </div>
  )
}

export default ContainerMaps