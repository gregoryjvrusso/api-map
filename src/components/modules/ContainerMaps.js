import React, { useState, useEffect } from 'react'
import GoogleMaps from '../elements/GoogleMaps/GoogleMaps'
import { geoCodeGenerator } from '../helpers/geocode'
import styled from 'styled-components'
import Text from '../elements/Text/Text'

const Container = styled.div`
  margin-top: 1em;
  max-height: 50%;
`
const HeaderContainer = styled.div`
  display: flex;
`
const CloseContainer = styled.div`
  min-width: 20%;
`
const Close = styled.p`
  font-size: 1.5em;
  font-weight: bold;
  text-align: right;
  color: ${p => p.theme.colors.backgrounds.primary.grayDark};
  margin-top: 0;
  cursor: pointer;
`
const ContainerTitle = styled.div`
  min-width: 80%;
`

const ContainerMaps = ({ data, setData }) => {
  const [latMap, setLat] = useState('')
  const [lngMap, setLng] = useState('')
  useEffect(() => {
    geoCodeGenerator(data.cep, { setLat }, { setLng })
  }, [data])

  const closeMap = () => {
    setData('')
  }

  return (
    <Container>
      <HeaderContainer>
        <ContainerTitle>
          <Text 
            text={data.logradouro} 
            fontSize='2em'
            fontWeight='bold'
            margin='0'
          />
        </ContainerTitle>
        <CloseContainer>
          <Close onClick={closeMap}>
            X
          </Close>
        </CloseContainer>
      </HeaderContainer>
      <Text text={data.bairro} margin='0' />
      <Text text={`${data.localidade}-${data.uf}`} margin='0' />
      <Text text={data.cep} margin='0 0 1em' />
      {lngMap !== '' && latMap !== '' &&
        <GoogleMaps
          latMap={latMap}
          lngMap={lngMap}
        />}
    </Container>
  )
}

export default ContainerMaps