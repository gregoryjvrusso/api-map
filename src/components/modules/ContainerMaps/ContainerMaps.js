import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import GoogleMaps from '../../elements/GoogleMaps/GoogleMaps'
import { ReactComponent as Close } from '../../elements/Icons/close.svg'
import { geoCodeGenerator } from '../../helpers/geocode'
import ContainerAdress from '../ContainterAdress/ContainerAdress'

const Container = styled.div`
  margin-top: 1em;
`
const HeaderContainer = styled.div`
  display: flex;
`
const CloseContainer = styled.div`
  min-width: 20%;
  display: flex;
  justify-content: flex-end;
`
const CloseElement = styled(Close)`
  cursor: pointer;
  padding: 1em;
`


const ContainerMaps = ({ data, setData }) => {
  const [latMap, setLat] = useState('')
  const [lngMap, setLng] = useState('')
  useEffect(() => {
    geoCodeGenerator(data.cep, { setLat }, { setLng })
  }, [data])

  const closeMap = () => {
    setData('')
  }

  return (
    <Container>
      <HeaderContainer>
        <ContainerAdress data={data} />
        <CloseContainer>
          <CloseElement onClick={closeMap} />
        </CloseContainer>
      </HeaderContainer>
      {lngMap !== '' && latMap !== '' &&
        <GoogleMaps
          latMap={latMap}
          lngMap={lngMap}
        />}
    </Container>
  )
}

export default ContainerMaps