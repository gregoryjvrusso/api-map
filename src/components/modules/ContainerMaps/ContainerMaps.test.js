import React from 'react'
import renderer from 'react-test-renderer'
import ContainerMaps from './ContainerMaps.js'

const mock = {
  "cep": "01001-000",
  "logradouro": "Praça da Sé",
  "complemento": "lado ímpar",
  "bairro": "Sé",
  "localidade": "São Paulo",
  "uf": "SP",
  "unidade": "",
  "ibge": "3550308",
  "gia": "1004"
}

describe('Container Maps component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
      .create(<ContainerMaps data={mock} />)
      .toJSON()
    expect(render).toMatchSnapshot()
  })
})
