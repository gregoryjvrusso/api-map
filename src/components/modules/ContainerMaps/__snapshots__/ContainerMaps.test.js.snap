// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Maps component deve conseguir renderizar [snapshot] 1`] = `
<div
  className="sc-htpNat gbqgVN"
>
  <div
    className="sc-bxivhb cDFCAd"
  >
    <div
      className="sc-bwzfXH wqKLy"
    >
      <p
        className="sc-bdVaJa fTxPHd"
        fontSize="2em"
        fontWeight="bold"
      >
        Praça da Sé
      </p>
      <p
        className="sc-bdVaJa ffYDP"
      >
        Sé
      </p>
      <p
        className="sc-bdVaJa ffYDP"
      >
        São Paulo-SP
      </p>
      <p
        className="sc-bdVaJa dqUOud"
      >
        01001-000
      </p>
    </div>
    <div
      className="sc-ifAKCX OJlAs"
    >
      <svg
        className="sc-EHOje iOXpbg"
        onClick={[Function]}
      >
        close.svg
      </svg>
    </div>
  </div>
</div>
`;
